extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authentication/endpoint/CoordinatorBasicAuthenticatorResourceHandler.java

```diff
@@ -0,0 +1,192 @@
+    if (authenticator == null) {
+      return makeResponseForAuthenticatorNotFound(authenticatorName);
+    }
+
+    Map<String, BasicAuthenticatorUser> userMap = storageUpdater.deserializeUserMap(
+        storageUpdater.getCurrentUserMapBytes(authenticatorName)
+    );
+
+    try {
+      BasicAuthenticatorUser user = userMap.get(userName);
+      if (user == null) {
+        throw new BasicSecurityDBResourceException("User [%s] does not exist.", userName);
+      }
+      return Response.ok(user).build();
+    }
+    catch (BasicSecurityDBResourceException cfe) {
+      return makeResponseForBasicSecurityDBResourceException(cfe);
+    }
+  }
+
+  @Override
+  public Response createUser(String authenticatorName, String userName)
+  {
+    final BasicHTTPAuthenticator authenticator = authenticatorMap.get(authenticatorName);
+    if (authenticator == null) {
+      return makeResponseForAuthenticatorNotFound(authenticatorName);
+    }
+
+    try {
+      storageUpdater.createUser(authenticatorName, userName);
+      return Response.ok().build();
+    }
+    catch (BasicSecurityDBResourceException cfe) {
+      return makeResponseForBasicSecurityDBResourceException(cfe);
+    }
+  }
+
+  @Override
+  public Response deleteUser(String authenticatorName, String userName)
+  {
+    final BasicHTTPAuthenticator authenticator = authenticatorMap.get(authenticatorName);
+    if (authenticator == null) {
+      return makeResponseForAuthenticatorNotFound(authenticatorName);
+    }
+
+    try {
+      storageUpdater.deleteUser(authenticatorName, userName);
+      return Response.ok().build();
+    }
+    catch (BasicSecurityDBResourceException cfe) {
+      return makeResponseForBasicSecurityDBResourceException(cfe);
+    }
+  }
+
+  @Override
+  public Response updateUserCredentials(String authenticatorName, String userName, String password)
+  {
+    final BasicHTTPAuthenticator authenticator = authenticatorMap.get(authenticatorName);
+    if (authenticator == null) {
+      return makeResponseForAuthenticatorNotFound(authenticatorName);
+    }
+
+    try {
+      storageUpdater.setUserCredentials(authenticatorName, userName, password.toCharArray());
+      return Response.ok().build();
+    }
+    catch (BasicSecurityDBResourceException cfe) {
+      return makeResponseForBasicSecurityDBResourceException(cfe);
+    }
+  }
+
+  @Override
+  public Response getCachedSerializedUserMap(String authenticatorName)
+  {
+    final BasicHTTPAuthenticator authenticator = authenticatorMap.get(authenticatorName);
+    if (authenticator == null) {
+      return makeResponseForAuthenticatorNotFound(authenticatorName);
+    }
+
+    return Response.ok(storageUpdater.getCachedSerializedUserMap(authenticatorName)).build();
+  }
+
+  @Override
+  public Response authenticatorUpdateListener(String authenticatorName, byte[] serializedUserMap)
+  {
+    return Response.status(Response.Status.NOT_FOUND).build();
```

Comment:
Maybe ```FORBIDDEN```.

Reply:
The coordinator/non-coordinator APIs are combined in one resource, because there wasn't a way for me to selectively create the resource based on node type in the extension module.

So I want to use NOT_FOUND, the idea being that the unsupported APIs should appear to the user as "not existing", rather than "exists but is forbidden"

Reply:
Thanks. Makes sense.